newIDE/app/src/AssetStore/AssetPackDialog.js

```diff
@@ -0,0 +1,212 @@
+          errorId: 'install-asset-pack-error',
+        });
+      } finally {
+        setIsAssetPackBeingInstalled(false);
+      }
+    },
+    [
+      resourcesFetcher,
+      eventsFunctionsExtensionsState,
+      project,
+      objectsContainer,
+      events,
+      onObjectAddedFromAsset,
+      onAssetPackAdded,
+    ]
+  );
+
+  return (
+    <Dialog
+      title={<Trans>{assetPack.name}</Trans>}
+      open
+      cannotBeDismissed
+      actions={[
+        <FlatButton
+          key="cancel"
+          label={<Trans>Cancel</Trans>}
+          disabled={isAssetPackBeingInstalled}
+          onClick={onClose}
+        />,
+        // Loading.
+        isAssetPackBeingInstalled && (
+          <FlatButton
+            key="loading"
+            label={<Trans>Please wait...</Trans>}
+            disabled
+            onClick={() => {}}
+          />
+        ),
+        // All assets already installed.
+        !isAssetPackBeingInstalled && missingAssetShortHeaders.length === 0 && (
+          <RaisedButton
+            key="install-again"
+            label={<Trans>Install again</Trans>}
+            primary={false}
+            onClick={() => onInstallAssetPack(assetShortHeaders)}
+          />
+        ),
+        // No assets installed.
+        !isAssetPackBeingInstalled &&
+          missingAssetShortHeaders.length === assetShortHeaders.length && (
+            <RaisedButton
+              key="continue"
+              label={<Trans>Continue</Trans>}
+              primary
+              onClick={() => onInstallAssetPack(assetShortHeaders)}
+            />
+          ),
+        // Some assets installed.
+        !isAssetPackBeingInstalled &&
+          missingAssetShortHeaders.length !== 0 &&
+          missingAssetShortHeaders.length !== assetShortHeaders.length && (
+            <RaisedButtonWithSplitMenu
+              label={<Trans>Install the missing assets</Trans>}
+              key="install-missing"
+              primary
+              onClick={() => {
+                onInstallAssetPack(missingAssetShortHeaders);
+              }}
+              buildMenuTemplate={i18n => [
+                {
+                  label: i18n._(t`Install all the assets`),
+                  click: () => onInstallAssetPack(assetShortHeaders),
+                },
+              ]}
+            />
+          ),
+      ]}
+      onApply={() => {
+        missingAssetShortHeaders.length === assetShortHeaders.length
+          ? onInstallAssetPack(assetShortHeaders)
+          : onInstallAssetPack(missingAssetShortHeaders);
+      }}
+    >
+      <Column>
+        {isAssetPackBeingInstalled ? (
+          <>
+            <Text>
+              <Trans>Installing assets...</Trans>
+            </Text>
+            <Line expand>
+              <LinearProgress style={styles.linearProgress} />
+            </Line>
+          </>
+        ) : (
+          <Text>
+            {missingAssetShortHeaders.length === 0 ? (
+              <Trans>
```

Comment:
The conditions are the same here. They are not very hard to understand but maybe boolean constants will be easier to read because it allows to use `||` and to jump between the 2 parts of the UI by highlighting them.